### ISSUE TITLE
[generic-extras] configured derivation prone to using wrong import

### ISSUE THREAD

**Issue by felixbr** [circe/circe#1220]
Hi,

My colleagues and I use `circe-generic-extras` in several projects and it's a really useful. There is, however, a recurring scenario where someone is using it and the code just doesn't seem to do the expected thing.

```
sealed trait Foo
object Foo {
  case object Bar extends Foo
  case object Baz extends Foo

  // sometimes imported from somewhere else
  implicit val derivationConfig: Configuration = 
    Configuration.default.withDiscriminator("__typename")

  implicit val encodeFoo: Encoder[Foo] = deriveEncoder
}
```

Most of the time people think the implicit `Configuration` is not being picked up correctly, so they move it around and try all sorts of things.

The real problem usually is that they imported
`io.circe.generic.semiauto._` instead of `io.circe.generic.extras.semiauto._`. The code still compiles and runs but it does the wrong thing.

In some cases it can help to activate `-Ywarn-unused:implicits` but only if the `Configuration` wasn't used somewhere else already. (And it's still a mental leap from "unused implicit" to "my circe imports are wrong")

I know this is a long shot as it would be a breaking change, but I think renaming `deriveEncoder` and `deriveDecoder` in `circe-generic-extras` to `deriveConfiguredEncoder` and `deriveConfiguredDecoder` respectively would make this much harder to get wrong (especially for people new to circe).
If a breaking change will not be considered I'd also be fine with deprecating it while introducing the new alias.

I'd love to hear your thoughts on this :)

Cheers
~ Felix

**Comment by travisbrown** [comment]
I agree that this is confusing, and personally I'd be happy to consider a PR that adds new `deriveConfiguredX` methods in generic-extras and deprecates the `deriveX` ones (I think we'll want to get more feedback before merging something like this, but there's a chance it could still happen in 0.12.0).

**Comment by travisbrown** [comment]
I've been thinking a little more about this, and I'm convinced we should do it. Would you be able to open a PR?

**Comment by felixbr** [comment]
I'll try to write one in the next days 👍

**Comment by travisbrown** [comment]
Fixed in #1221 and will be released in an 0.12.0-RC3 this morning (probably).
